htdocs/core/class/commonobject.class.php

```diff
@@ -6767,7 +6767,7 @@ public function showOptionals($extrafields, $mode = 'view', $params = null, $key
 					// Convert float submited string into real php numeric (value in memory must be a php numeric)
 					if (in_array($extrafields->attributes[$this->table_element]['type'][$key], array('price', 'double')))
 					{
-						$value = GETPOSTISSET($keyprefix.'options_'.$key.$keysuffix) ?price2num(GETPOST($keyprefix.'options_'.$key.$keysuffix, 'alpha', 3)) : $this->array_options['options_'.$key];
+						$value = (!empty($value)) ? price2num($value) : $this->array_options['options_'.$key];
```

Comment:
This looks strange. How do you set value of price to 0 or to empty after this change ?

Reply:
Yes, the field was not initialised with default value. I aknowledge we also need to check if GETPOSTISSET which should allow to set to 0 or empty.